tests/base.py

```diff
@@ -314,15 +310,31 @@ def split_records_into_created_and_updated(self, records):
                                    'schema': batch['schema'],
                                    'key_names' : batch.get('key_names'),
                                    'table_version': batch.get('table_version')}
-            created[stream]['messages'] += [m for m in batch['messages']
-                                                if m['data'].get("updated") == m['data'].get(bookmark_key)]
+            # Bookmark key changed for `invoices` from `date` to `created` due to latest API change
+            # but for `invoices` stream, the `created` field have integer type(epoch format) from starting so
+            # converting `updated` to epoch for comparison.
+            if stream == "invoices":
+                created[stream]['messages'] += [m for m in batch['messages']
+                                                    if self.dt_to_ts(m['data'].get("updated")) == m['data'].get(bookmark_key)]
+            else:
+                created[stream]['messages'] += [m for m in batch['messages']
+                                                    if m['data'].get("updated") == m['data'].get(bookmark_key)]
+
             if stream not in updated:
                 updated[stream] = {'messages': [],
                                    'schema': batch['schema'],
                                    'key_names' : batch.get('key_names'),
                                    'table_version': batch.get('table_version')}
-            updated[stream]['messages'] += [m for m in batch['messages']
-                                                if m['data'].get("updated") != m['data'].get(bookmark_key)]
+
+            # Bookmark key changed for `invoices` from `date` to `created` due to latest API change
+            # but for `invoices` stream, the `created` field have integer type(epoch format) from starting so
+            # converting `updated` to epoch for comparison.
+            if stream == "invoices":
+                updated[stream]['messages'] += [m for m in batch['messages']
+                                                    if self.dt_to_ts(m['data'].get("updated")) != m['data'].get(bookmark_key)]
+            else:
+                updated[stream]['messages'] += [m for m in batch['messages']
+                                                    if m['data'].get("updated") != m['data'].get(bookmark_key)]
```

Comment:
I'm little bit confused here. In the previous section we are converting updated to integer type to compare with created which made sense. Why are we here doing the comparison with Updated and Updated?

Reply:
We are doing similar things in the previous section and above section. The only difference is in the previous section we are checking with the `==` operator to get records that are created. While in the above section we are checking with the `!=` operator to get records that are updated.